### PR TITLE
DOC/API: Move Styler import

### PR DESCRIPTION
**Pull request by TomAugspurger**
From the top-level to pandas.io.formats.style.Styler, which is a bit wordy
but I was having circular-import issues at pandas.io.formats.
I think that's ok since people won't really be using this interactively.

Closes https://github.com/pandas-dev/pandas/issues/16009

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
api.rst also needs an update

**Comment by jorisvandenbossche** [comment]
In this issue (https://github.com/pandas-dev/pandas/issues/16009), I also raised the question whether we should deprecate the old location. But is there currently a reason that a user would use the actual object instead of access it through `df.style` ?

**Comment by TomAugspurger** [comment]
I believe api.rst was already changed as part of the move.

```
Style
-----
.. currentmodule:: pandas.io.formats.style

``Styler`` objects are returned by :attr:`pandas.DataFrame.style`.
```

For the deprecation... I could go either way. 5516aa4 adds a compatibility shim pointing to the new location. I'll add it to the master list if we decide to include it.

**Comment by jorisvandenbossche** [comment]
> I believe api.rst was already changed as part of the move.

Ah yes, my bad! (I actually even said it myself https://github.com/pandas-dev/pandas/issues/16009#issuecomment-294867514 :-))

**Comment by TomAugspurger** [comment]
@jorisvandenbossche all good for you? Is everyone OK with a deprecation warning from `pd.core.formats.s, to be removed in 0.21?
